controllers/grid/files/form/ManageSubmissionFilesForm.inc.php

```diff
@@ -101,9 +103,14 @@ function execute($args, $request, $stageSubmissionFiles, $fileStage) {
 	 * Determine if a file is already present in the stage.
 	 * @param $submissionFile SubmissionFile The submission file
 	 * @param $stageSubmissionFiles array The list of submission files in the stage.
+	 * @param $fileStage int FILE_STAGE_...
 	 */
-	protected function _fileExistsInStage($submissionFile, $stageSubmissionFiles) {
-		return array_key_exists($submissionFile->getFileId(), $stageSubmissionFiles);
+	protected function _fileExistsInStage($submissionFile, $stageSubmissionFiles, $fileStage) {
+		if (!isset($stageSubmissionFiles[$submissionFile->getFileId()])) return false;
+		foreach ($stageSubmissionFiles[$submissionFile->getFileId()] as $stageFile) {
+			if ($stageFile->getFileStage() == $submissionFile->getFileStage() && $stageFile->getFileStage() == $fileStage) return true;
+		}
+		return false;
```

Comment:
You changed this to make possible to move files between file stages and not only workflow stages right? If yes, I think we could make it clearer in the method doc, adding file before the stage word